### PR TITLE
fix(otelglobal): fix multiple set attributes calls overwriting existing span attributes

### PR DESCRIPTION
**Pull request by asecondo**
Fixes issues described in #592

### CODE REVIEW

File: .github/workflows/e2e/k8s/sample-job.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ spec:
           value: "tracecontext,baggage"
         - name: OTEL_GO_AUTO_INCLUDE_DB_STATEMENT
           value: "true"
+        - name: OTEL_GO_AUTO_SHOW_VERIFIER_LOG
+          value: "true"
```
Comment (asecondo) [inline .github/workflows/e2e/k8s/sample-job.yml:42]: chore: remove this

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ OpenTelemetry Go Automatic Instrumentation adheres to [Semantic Versioning](http
 
 - Log any failures to close running probes. ([#586](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/586))
 - Log explanatory error message on Linux kernel lockdown ([#290](https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/290))
+- (otelglobal) Fixed case where multiple span.SetAttributes() calls would overwrite existing attributes ([](TODO(NOW): fill in PR once it's available))
```
Comment (asecondo) [inline CHANGELOG.md:23]: chore: update this next commit

File: internal/pkg/instrumentation/utils/ebpf.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func LoadEBPFObjects(spec *ebpf.CollectionSpec, to interface{}, opts *ebpf.Colle
 	showVerifierLogs := shouldShowVerifierLogs()
 	if showVerifierLogs {
 		opts.Programs.LogSize = ebpf.DefaultVerifierLogSize * 100
-		opts.Programs.LogLevel = ebpf.LogLevelInstruction | ebpf.LogLevelBranch | ebpf.LogLevelStats
+		opts.Programs.LogLevel = /*ebpf.LogLevelInstruction | ebpf.LogLevelBranch |*/ ebpf.LogLevelStats
```
Comment (asecondo) [inline internal/pkg/instrumentation/utils/ebpf.go:38]: chore: revert this

File: internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/bpf/probe.bpf.c [hunk 4]
```diff
@@ -143,6 +143,8 @@ int uprobe_Start_Returns(struct pt_regs *ctx) {
         otel_span->sc = generate_span_context();
     }
 
+    otel_span->attributes.valid_attrs = 0;
```
Comment (asecondo) [inline internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/bpf/probe.bpf.c:146]: chore: don't think this is actually needed. should remove.

File: internal/include/otel_types.h [hunk 5]
```diff
@@ -102,13 +102,16 @@ static __always_inline void convert_go_otel_attributes(void *attrs_buf, s64 slic
...
 	go_otel_attr_value_t go_attr_value = {0};
 	struct go_string go_str = {0};
-	u8 valid_attrs = 0;
+	u8 valid_attrs = enc_attrs->valid_attrs;
+	if (valid_attrs >= OTEL_ATTRUBUTE_MAX_COUNT) {
+		return;
+	}
 
-	for (u32 go_attr_index = 0; go_attr_index < num_attrs; go_attr_index++) {
+	for (u8 go_attr_index = 0; go_attr_index < num_attrs; go_attr_index++) {
```
Comment (asecondo) [inline internal/include/otel_types.h:114]: thought: just as a general note - this was really tricky to find. without downsizing `go_attr_index` and `num_attrs`, i was getting an ebpf error where the max number of instructions was being exceeded, which with the default logs you only see that the verifier prints a general "too many arguments" error. this eventually became more clear when i reduced the noise in the logs and only enabled `ebpf.LogLevelStats`.

Comment (RonFed) [reply internal/include/otel_types.h:114]: Right, we should probably change the default verifier log settings.

**Review by RonFed — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**